bug: `opencli --version` reports `0.0.0` when installed from source + `Failed to load module` for user clis

## Description

When installing from source (as documented in the README under **For Developers**), two issues occur simultaneously:

1. `opencli --version` always reports `0.0.0` instead of the actual version.
2. Every command in `~/.opencli/clis/` fails with `Failed to load module ... Cannot find module '/Users/<user>/.opencli/src/registry.js'`.

---

## Environment

- **Install method:** from source (`git clone` → `npm install` → `npm run build` → `npm link` / `pnpm link --global`)
- **Node:** v24.14.1 (via nvm)
- **npm:** latest
- **OS:** macOS
- **opencli version (package.json):** 1.6.3

---

## Steps to Reproduce

```bash
git clone git@github.com:jackwener/opencli.git && cd opencli
npm install
npm run build
npm link   # or: pnpm link --global
opencli --version
opencli doctor
```

---

## Root Cause Analysis

### Bug 1 — `version` reports `0.0.0`

`src/version.ts` resolves `package.json` with one level up from `__dirname`:

```ts
// src/version.ts
const pkgJsonPath = path.resolve(__dirname, '..', 'package.json');
```

At runtime after `npm run build`, `__dirname` is `dist/src/`. One level up is `dist/`, where **no `package.json` exists**. The `try/catch` silently falls back to `'0.0.0'`.

This works fine for `npm install -g` users because npm automatically includes the root `package.json` in the published package, placing it at `<package-root>/package.json` — exactly one level above `dist/src/`. But when running from a cloned repo, the root `package.json` is two levels above `dist/src/`.

**Fix:**

```diff
- const pkgJsonPath = path.resolve(__dirname, '..', 'package.json');
+ const pkgJsonPath = path.resolve(__dirname, '..', '..', 'package.json');
```

### Bug 2 — `Failed to load module ~/.opencli/clis/*.js`

The `postinstall` / `fetch-adapters.js` script downloads pre-compiled `.js` adapter files into `~/.opencli/clis/`. These files contain hardcoded absolute-style imports:

```
Cannot find module '/Users/<user>/.opencli/src/registry.js'
Cannot find module '/Users/<user>/.opencli/src/errors.js'
```

However, `~/.opencli/src/` is never created — the core runtime lives inside the npm package's `dist/src/`, not under `~/.opencli/`. These pre-compiled clis appear to have been built against an older layout where the source was symlinked or copied into `~/.opencli/src/`.

**Workaround (user-side):** `rm -rf ~/.opencli/clis/` causes opencli to fall back to the built-in YAML-driven commands in `dist/clis/`, avoiding the broken imports entirely.

---

## Expected Behavior

- `opencli --version` → `1.6.3`
- `opencli doctor` → no `Failed to load module` warnings

## Actual Behavior

- `opencli --version` → `0.0.0`
- Hundreds of `⚠ Failed to load module` warnings for every adapter in `~/.opencli/clis/`

---

## Suggested Fix

For Bug 1, the one-line change above in `src/version.ts` is sufficient. It is backward-compatible: when installed via `npm install -g`, the root `package.json` is still two levels above `dist/src/` inside the installed package directory.

For Bug 2, the pre-compiled `.js` files distributed by `fetch-adapters.js` may need to be rebuilt with correct import paths that resolve against the npm package's installed location rather than `~/.opencli/src/`.

--- a/src/discovery.ts
+++ b/src/discovery.ts
@@ -105,11 +105,16 @@ export async function ensureUserCliCompatShims(baseDir: string = USER_OPENCLI_DI
   const writes: Promise<void>[] = [];
 
   // Root-level shims (both with and without .js extension)
+  // Also generate src/ shims: adapters import ../../src/registry.js etc., which
+  // resolve to ~/.opencli/src/registry.js when running from ~/.opencli/clis/<site>/
+  const srcDir = path.join(baseDir, 'src');
+  await fs.promises.mkdir(srcDir, { recursive: true });
   for (const [shimName, moduleName] of rootShims) {
     const url = pathToFileURL(resolveHostRuntimeModulePath(moduleName)).href;
     const content = `export * from '${url}';\n`;
     writes.push(writeCompatShimIfNeeded(path.join(baseDir, shimName), content));
     writes.push(writeCompatShimIfNeeded(path.join(baseDir, `${shimName}.js`), content));
+    writes.push(writeCompatShimIfNeeded(path.join(srcDir, `${shimName}.js`), content));
   }
 
   // Subdirectory shims

--- a/src/version.ts
+++ b/src/version.ts
@@ -7,7 +7,14 @@ import * as path from 'node:path';
 import { fileURLToPath } from 'node:url';
 
 const __dirname = path.dirname(fileURLToPath(import.meta.url));
-const pkgJsonPath = path.resolve(__dirname, '..', 'package.json');
+
+// Dev: __dirname is src/ (one level to root).
+// Prod: __dirname is dist/src/ (two levels to root).
+let _pkgDir = path.resolve(__dirname, '..');
+if (!fs.existsSync(path.join(_pkgDir, 'package.json'))) {
+  _pkgDir = path.resolve(_pkgDir, '..');
+}
+const pkgJsonPath = path.join(_pkgDir, 'package.json');
 
 export const PKG_VERSION: string = (() => {
   try {

